src/update/UpdateManager.cpp

```diff
@@ -91,12 +97,16 @@ void UpdateManager::GetUpdates(bool ignoreNetworkErrors)
     response->setRequestFinishedCallback([response, ignoreNetworkErrors, this](audacity::network_manager::IResponse*) {
         
         // We don't' want to duplicate the updates checking if that already launched.
-        if(!mUpdateMutex.try_lock())
         {
-            response->abort();
-            return;
+            std::lock_guard<std::mutex> lock(mUpdateMutex);
```

Comment:
You know you can leave out `<std::mutex>` in C++17?  Improved deduction of types in constructor expressions and declarations!

I am convinced now that this mutex is doing a correct thing.  I think there would be a more ligthweight way to do this with std::atomic instead -- the mutex is only guarding the transition of a boolen from false to true.  But, don't bother doing this.

Reply:
But -- couldn't the setting of mOnProgress to true be done sooner?  Before the request is even made?